src/flow/domain/single.rs

```diff
@@ -161,7 +161,7 @@ impl NodeDescriptor {
             }
             flow::node::Type::Internal(ref mut i) => {
                 let from = m.link().src;
-                m.map_data(|data| i.on_input(from, data, nodes, state));
+                m.map_data(|data| i.on_input(from, data, nodes, state).unwrap());
```

Comment:
Well, `unwrap` indicates that you believe `on_input` can *never* return `None` in this context. Why is that?